decidim-core/app/controllers/decidim/newsletters_controller.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module Decidim
+  # The controller to show the newsletter on the website.
+  class NewslettersController < Decidim::ApplicationController
+    skip_authorization_check
+
+    layout "decidim/mailer", only: [:show]
+    helper Decidim::SanitizeHelper
+    include Decidim::NewslettersHelper
+
+    helper_method :newsletter
+
+    def show
+      @user = current_user
+      @organization = current_organization
+
+      if newsletter.sent?
+        if @user.present?
+          @encrypted_token = Decidim::NewsletterEncryptor.sent_at_encrypted(@user.id, newsletter.sent_at)
+        end
+        @body = parse_interpolations(newsletter.body[I18n.locale.to_s], @user, newsletter.id)
+      else
+        redirect_to decidim.root_url(host: @organization.host)
+      end
+    end
+
+    def unsubscribe
+      encryptor = Decidim::NewsletterEncryptor
+
+      decrypted_string = encryptor.sent_at_decrypted(params[:u])
+      user = User.find_by(decidim_organization_id: current_organization.id, id: decrypted_string.split("-").first)
+      sent_at_time = Time.zone.at(decrypted_string.split("-").second.to_i)
+
+      if sent_at_time > (Time.current - 15.days)
```

Comment:
Why do we need a time limit here?

Reply:
as @josepjaume said few days ago, the encrypted_token includes a timestamp to make that token invalid after two weeks.

Reply:
Oh, OK, talked offline with him.